src/averageAges.js

```diff
@@ -59,6 +80,23 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+  const MOTHERS = people.filter(mom =>
+    people.some(child => child.mother === mom.name));
+
+  const CHILDREN_WITH_MOTHERS = people.filter(child =>
+    (!onlyWithSon && MOTHERS.some(mom => mom.name === child.mother))
+    || (onlyWithSon && child.sex === 'm'
+    && MOTHERS.some(mom => mom.name === child.mother))
+  );
```

Comment:
Try to simplify this
<img width="1048" alt="image" src="https://github.com/mate-academy/js_average-ages/assets/86789161/f57babb5-893b-4c83-a916-1a3880901145">